src/evict/evict_lru.c

```diff
@@ -2164,6 +2192,8 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
         __wt_verbose(session, WT_VERB_EVICTSERVER, "walk select: %p, size %" WT_SIZET_FMT,
           (void *)page, page->memory_footprint);
     }
+    if (F_ISSET(cache, WT_CACHE_EVICT_DIRTY_HARD | WT_CACHE_EVICT_UPDATES_HARD))
```

Comment:
currently, the eviction state is not changed outside, so these values don't change from the start to the end of the function, but to make it future-proof, set a local variable whether there is a snapshot and based on it free the snapshot would be good. This way in case in future if the eviction state gets changed concurrently, we will not get any problem in freeing the snapshot.

Reply:
done